src/scope/core/pipelines/registry.py

```diff
@@ -0,0 +1,103 @@
+"""Pipeline registry for centralized pipeline management.
+
+This module provides a registry pattern to eliminate if/elif chains when
+accessing pipelines by ID. It enables dynamic pipeline discovery and
+metadata retrieval.
+"""
+
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from .interface import Pipeline
+    from .schema import BasePipelineConfig
+
+
+class PipelineRegistry:
+    """Registry for managing available pipelines."""
+
+    _pipelines: dict[str, type["Pipeline"]] = {}
+
+    @classmethod
+    def register(cls, pipeline_id: str, pipeline_class: type["Pipeline"]) -> None:
+        """Register a pipeline class with its ID.
+
+        Args:
+            pipeline_id: Unique identifier for the pipeline
+            pipeline_class: Pipeline class to register
+        """
+        cls._pipelines[pipeline_id] = pipeline_class
+
+    @classmethod
+    def get(cls, pipeline_id: str) -> type["Pipeline"] | None:
+        """Get a pipeline class by its ID.
+
+        Args:
+            pipeline_id: Pipeline identifier
+
+        Returns:
+            Pipeline class if found, None otherwise
+        """
+        return cls._pipelines.get(pipeline_id)
+
+    @classmethod
+    def get_config_class(cls, pipeline_id: str) -> type["BasePipelineConfig"] | None:
+        """Get config class for a specific pipeline.
+
+        Args:
+            pipeline_id: Pipeline identifier
+
+        Returns:
+            Pydantic config class if found, None otherwise
+        """
+        pipeline_class = cls.get(pipeline_id)
+        if pipeline_class is None:
+            return None
+        return pipeline_class.get_config_class()
+
+    @classmethod
+    def get_schema(cls, pipeline_id: str) -> dict[str, Any] | None:
```

Comment:
If a caller can get the schema from the return value of get_config_class() we can omit this.